src/classes/RD2_VisualizeScheduleController_TEST.cls

```diff
@@ -170,45 +223,250 @@ private with sharing class RD2_VisualizeScheduleController_TEST {
+        System.assertEquals(dayOfMonth, schedules[0].dayOfMonth, 'Day of month should be 12.');
+    }
+
+    /****
+    * @description Validates that 2 active schedule are returned for an RD with current and future schedules
+    */
+    @IsTest
+    private static void shouldReturnActiveSchedulesWhenRDHasCurrentAndFutureSchedule() {
+        Date startDate = Date.newInstance(2018,2,1);
+        Date newStartDate = Date.newInstance(2018,4,1);
+        Date serviceDate = Date.newInstance(2018,3,1);
+        Date endDate = Date.newInstance(2018,3,31);
+        String dayOfMonth = '12';
+        npe03__Recurring_Donation__c rd = getRecurringDonationMonthlyBuilder()
+                .withStartDate(startDate)
+                .withDayOfMonth(dayOfMonth)
+                .build();
+
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Test.startTest();
+        insert rd;
+        rd.StartDate__c = newStartDate;
+        rd.npe03__Amount__c = 200;
+        update rd;
+        setScheduleServiceDate(serviceDate);
+        List<RD2_VisualizeScheduleController.Schedule> schedules = RD2_VisualizeScheduleController.getSchedules(rd.Id);
+        Test.stopTest();
+
+        System.assertEquals(2, schedules.size(), 'There should be 2 schedules.');
+        System.assertEquals(true, schedules[0].isCurrent, 'isCurrent should be true.');
+        System.assertEquals(false, schedules[1].isCurrent, 'isCurrent should be false.');
+        System.assertEquals(100, schedules[0].amount, 'Amount should be 100.');
+        System.assertEquals(200, schedules[1].amount, 'Amount should be 200.');
+        System.assertEquals(startDate, schedules[0].startDate, 'Start Date should be 2/1/2018.');
+        System.assertEquals(newStartDate, schedules[1].startDate, 'Start Date should be 4/1/2018.');
+        System.assertEquals(endDate, schedules[0].endDate, 'End Date should be 3/31/2018.');
+        System.assertEquals(null, schedules[1].endDate, 'End Date should be null.');
+        System.assertEquals(RD2_Constants.INSTALLMENT_PERIOD_MONTHLY, schedules[0].period, 'Period should be monthly.');
+        System.assertEquals(RD2_Constants.INSTALLMENT_PERIOD_MONTHLY, schedules[1].period, 'Period should be monthly.');
+        System.assertEquals(dayOfMonth, schedules[0].dayOfMonth, 'Day of month should be 12.');
+        System.assertEquals(dayOfMonth, schedules[1].dayOfMonth, 'Day of month should be 12.');
+    }
+
+    /****
+    * @description Validates that zero active schedules are returned for a closed RD
+    */
+    @IsTest
+    private static void shouldThrowAuraHandledExceptionRDIsClosed() {
+        npe03__Recurring_Donation__c rd = getRecurringDonationMonthlyBuilder()
+                .withStatusClosed()
+                .withDayOfMonth('12')
+                .build();
+
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Test.startTest();
+        insert rd;
+        Test.stopTest();
+
+        try {
```

Comment:
Let's checkout `actualException` in this test instead of asserting in the `try` block and verify exception in the `catch` block.